components/NumberInput.js

```diff
@@ -13,11 +13,22 @@ class NumberInput extends Component {
     this.setState(Object.assign({}, this.state, {value: parseInt(ev.target.value)}))
   }
 
-  blur(ev) {
+  _finishEditing() {
```

Comment:
Yeah, I noticed a distinct lack of semicolons and a new (a,b) => {...} syntax in your closures.  In general I'll just try to make things look like what's already there and maybe I'll pick up some new syntax and style in the process.